crates/aptos/src/move_tool/lint.rs

```diff
@@ -0,0 +1,65 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use super::IncludedArtifactsArgs;
+use crate::common::types::{CliCommand, CliError, CliTypedResult, MovePackageDir};
+use aptos_framework::{BuildOptions, BuiltPackage};
+use async_trait::async_trait;
+use clap::Parser;
+use move_compiler_v2::Experiment;
+use move_model::metadata::CompilerVersion;
+
+/// Run a Lint tool to show additional warnings, in addition to ordinary warnings and/or errors
+/// generated by the Move 2 compiler.  Currently this is done by running the Move 2 compiler with a
+/// special pass, so some other compiler options are also available, but in the future this may be a
+/// dedicated tool.
+///
+/// Ideally, we would skip code generation, but currently this is not possible. We do omit
+/// metadata generation, which hopefully renders the output not able to be uploaded.
+#[derive(Parser)]
+pub struct LintPackage {
+    // TODO: add some options to select certain lint/warning passes?
+    #[clap(flatten)]
+    pub(crate) included_artifacts_args: IncludedArtifactsArgs,
+
+    #[clap(flatten)]
+    pub(crate) move_options: MovePackageDir,
+}
+
+#[async_trait]
+impl CliCommand<&'static str> for LintPackage {
+    fn command_name(&self) -> &'static str {
+        "LintPackage"
+    }
+
+    async fn execute(self) -> CliTypedResult<&'static str> {
+        let move_options = MovePackageDir {
+            move_2: true,
```

Comment:
When I do `aptos move lint --help`, it shows that:
* I can specify the compiler version with `--compiler-version`. Do we want to expose this to the user? It also incorrectly says that the compiler version defaults to v1 when running the `aptos move lint`, but this is not true.
* I can specify the optimization level with `--optimization-level`. Do we want to expose this to the user?

Reply:
I can also specify `--bytecode-version` (the version compiler is going to emit). This also does not make sense to expose to the user in the context of the linter.

Reply:
I mention that `lint` defaults to 2.  Someone might want to try lint with --optimize=none.

Reply:
Lint now has its own options struct, so that's gone.

Reply:
Awesome!